File meta data persistent storage and bloom filters functions added.

acekillersg opened the pull request:
1. File meta data can be written to HDFS with native HDFSBolt implementation;
2. Implement bloom filters for all Java primitive types and can be easily extended to user-defined classes;
3. StormHDFSTest.class and BloomFilterTest.class are two classes to test 1. and 2., repectively;
4. More detailed instructions on how to use them can be found in README.md.

acekillersg committed:
Change "Texi" to "Taxi"

acekillersg committed:
Change "Texi" to "Taxi"

acekillersg committed:
Change "Texi" to "Taxi"

acekillersg committed:
Add LoggingTable to append meta logs to HDFS or reconstruct a RTree from HDFS data:
1) LoggingTable.append(): a static method which can be called anywhere to append a FileMetaData item to HDFS files for persistent storing;
2) LoggingTable.reconstruct(): a static method that reconstruct a new RTree from HDFS files and return the new constructed RTree.

Add some macros to TopologyConfig class which indicate local HDFS path. They should be replaced by real path in cluster.

acekillersg committed:
Create .gitignore

acekillersg committed:
Delete IndexingTopology.iml

acekillersg committed:
add some .gitignore items

acekillersg committed:
Remove .idea folder

acekillersg committed:
Add a RTree return function maintained by LoggingTable itself.

acekillersg committed:
Add a RTree return function to get the latest RTree maintained by LoggingTable class itself.

acekillersg committed:
Add a RTree return function to get the latest RTree maintained by LoggingTable class itself.

acekillersg committed:
Add a RTree return function to get the latest RTree maintained by LoggingTable class itself.

acekillersg committed:
Add a RTree return function to get the latest RTree maintained by LoggingTable class itself.

acekillersg committed:
Add bloom filter to Car and City class. However, this can be easily extended to any other data types or classes if necessary.

acekillersg committed:
Add bloom filter to Car and City class. However, this can be easily extended to any other data types or classes if necessary.

acekillersg committed:
Add bloom filter to Car and City class. However, this can be easily extended to any other data types or classes if necessary.

acekillersg committed:
1. Implemented bloom filters for all Java primitive types;
2. add a bloom filter testing class.
3. Detailed usage of bloom filter can be found in README.md.

wangli1426 commented:
Hi @acekillersg,

Thanks for the poll request. Before I merging this poll request, could you please address the following comments?

1. Since this readme is for the entire project, can you put the readme about the bloom filter and HDFSBolt in other files rather the readme.md in the root folder.

2. In your last pull request, I found some large but unnecessary files in your commit history. Did you remove the files from the history?

3. There are too many entries in .gitignore file. For example, all the lines starting with ```.idea/``` can be simply replaced by a single line ```.idea```

Many thanks.

acekillersg committed:
1. Implemented bloom filters for all Java primitive types;
2. add a bloom filter testing class.
3. Detailed usage of bloom filter can be found in README.md.

acekillersg commented:
Hi @wangli1426 ,

1) I have removed the README file and split it into bloom/ and metadata/ subfolders, respectively;
2) I have removed the .dat file and added it into .gitignore;
3) I have simplified the .gitignore file.

Thanks,
John

wangli1426 commented:
@acekillersg You did exactly what I want. Good work! Thank you.
